Std/Data/HashMap/WF.lean

```diff
@@ -321,7 +321,8 @@ theorem WF.filterMap {α β γ} {f : α → β → Option γ} [BEq α] [Hashable
     rw [← List.pairwise_map (R := (¬ · == ·))] at this ⊢
     exact this.sublist (H3 l.toList)
   · simp [Array.getElem_eq_data_get] at h ⊢
-    have := H.out.2.2 _ h; simp [AssocList.All] at this ⊢
+    have := H.out.2.2 _ h
+    simp [-List.mem_filterMap, AssocList.All] at this ⊢
```

Comment:
This looks vaguely suspicious, perhaps a `simpNF` issue?

Reply:
Yeah. `List.mem_filterMap` is simp in mathlib, but I'm not familiar enough with this proof nor with the list simp normal form to diagnose this.

Reply:
Okay, investigated -- the lemma is fine, the proof is the weird one here. The proof was trying to reason at the List level about filterMap instead of at the prop level using membership.